cdap-watchdog/src/test/java/co/cask/cdap/logging/logbuffer/recover/LogBufferRecoveryServiceTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.logbuffer.recover;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import co.cask.cdap.api.metrics.MetricsContext;
+import co.cask.cdap.api.metrics.NoopMetricsContext;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.logging.LoggingContext;
+import co.cask.cdap.common.utils.Tasks;
+import co.cask.cdap.logging.appender.LogMessage;
+import co.cask.cdap.logging.context.WorkerLoggingContext;
+import co.cask.cdap.logging.logbuffer.LogBufferWriter;
+import co.cask.cdap.logging.logbuffer.MockCheckpointManager;
+import co.cask.cdap.logging.pipeline.LogPipelineTestUtil;
+import co.cask.cdap.logging.pipeline.LogProcessorPipelineContext;
+import co.cask.cdap.logging.pipeline.MockAppender;
+import co.cask.cdap.logging.pipeline.logbuffer.LogBufferPipelineConfig;
+import co.cask.cdap.logging.pipeline.logbuffer.LogBufferProcessorPipeline;
+import co.cask.cdap.logging.serialize.LoggingEventSerializer;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Tests for {@link LogBufferRecoveryService}.
+ */
+public class LogBufferRecoveryServiceTest {
```

Comment:
Would be good to add a test to make sure that file cleanup does not start until recovery is complete

Reply:
Ok, filed https://issues.cask.co/browse/CDAP-15075